projects/igniteui-angular/src/lib/slider/thumb/thumb-slider.component.ts

```diff
@@ -0,0 +1,197 @@
+import {
+    NgModule,
+    Component,
+    Input,
+    HostListener,
+    ElementRef,
+    HostBinding,
+    Output,
+    EventEmitter,
+    OnInit,
+    OnDestroy} from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { Observable, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+@Component({
+    selector: 'igx-thumb',
+    templateUrl: 'thumb-slider.component.html'
+})
+export class IgxSliderThumbComponent implements OnInit, OnDestroy {
+
+    private _timer;
+    private _isActiveLabel = false;
+    private _destroy$ = new Subject<boolean>();
+
+    public isActive = false;
+
+    @Input()
+    public value: number;
+
+    @Input()
+    public continuous: boolean;
+
+    @Input()
+    public thumbLabelVisibilityDuration;
+
+    @Input()
+    public disabled: boolean;
+
+    @Input()
+    public onPan: Subject<number>;
+
+    @Input()
+    public stepDistance: number;
+
+    @Input()
+    public step: number;
+
+    @Output()
+    public onThumbValueChange = new EventEmitter<number>();
+
+    @HostBinding('attr.tabindex')
+    public tabindex = 0;
+
+    @HostBinding('class.igx-slider__thumb-to')
```

Comment:
Class name have to change to igx-slider__thumb, because it same for to and from thumb